### PR TITLE
[Java] maven plugin to clean-up output before generation

### PR DESCRIPTION
**Pull request by borsch**
https://github.com/OpenAPITools/openapi-generator/issues/14928

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [ ] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

[@wing328](https://github.com/wing328) (2015/07) [❤️](https://www.patreon.com/wing328)
    [@jimschubert](https://github.com/jimschubert) (2016/05) [❤️](https://www.patreon.com/jimschubert)
    [@cbornet](https://github.com/cbornet) (2016/05)
    [@jmini](https://github.com/jmini) (2018/04) [❤️](https://www.patreon.com/jmini)
    [@etherealjoy](https://github.com/etherealjoy) (2019/06)
    [@spacether](https://github.com/spacether) (2020/05) [❤️](https://github.com/sponsors/spacether/)

### CODE REVIEW

File: modules/openapi-generator-maven-plugin/src/main/java/org/openapitools/codegen/plugin/CodeGenMojo.java [hunk 1]
```diff
@@ -99,6 +100,9 @@ public class CodeGenMojo extends AbstractMojo {
     @Parameter(name = "output", property = "openapi.generator.maven.plugin.output")
     private File output;
 
+    @Parameter(name = "cleanUpOutputFolder", property = "openapi.generator.maven.plugin.cleanUpOutputFolder", defaultValue = "false")
```
Comment (wing328) [inline modules/openapi-generator-maven-plugin/src/main/java/org/openapitools/codegen/plugin/CodeGenMojo.java:103]: what about naming the option as `cleanupOutput` to make it consistent with the gradle plugin?

ref: https://github.com/OpenAPITools/openapi-generator/pull/13659

Comment (borsch) [reply modules/openapi-generator-maven-plugin/src/main/java/org/openapitools/codegen/plugin/CodeGenMojo.java]: @wing328 I've changed name

**Comment by wing328** [comment]
The CI tests (maven plugin) failed. Can you please take a look when you've time?

**Comment by borsch** [comment]
@wing328 issue was due to this change https://github.com/OpenAPITools/openapi-generator/commit/564939a4a8eb73f26a5878445c058875e2a1f2a9
Now jersey2 use diamond operator here https://github.com/OpenAPITools/openapi-generator/commit/564939a4a8eb73f26a5878445c058875e2a1f2a9#diff-e9ec55c3c14a0cce87cfa478732f202228aa39c395d1bae2c2f9dd7ea5074175R100 which is not supported inside maven plugin samples because java 1.7 was configured there

Is there a way to trigger maven/gradle plugin build once related libraries are updated? Like in this case to trigger jersey2 sample generation with maven plugin

**Comment by wing328** [comment]
> Is there a way to trigger maven/gradle plugin build once related libraries are updated? Like in this case to trigger jersey2 sample generation with maven plugin

it's possible. i'll see what i can do after merging this PR.

**Pull request merged**